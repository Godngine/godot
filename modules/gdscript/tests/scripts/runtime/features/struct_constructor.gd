struct TestStruct:
	var test_int : int = 1
	var test_float : float = 2.5
	var test_dict : Dictionary = {4 : 5}
	var test_array : Array = [3, "Godot"]

func test():
	print(TestStruct())
	print(TestStruct(3))
	print(TestStruct(3, 1.5))
	print(TestStruct(3, 1.5, {"key": "val"}))
	print(TestStruct(3, 1.5, {"key": "val"}, [0.5]))

	print(PropertyInfo())
	print(Object.PropertyInfo())
	# print(self.TestStruct())
