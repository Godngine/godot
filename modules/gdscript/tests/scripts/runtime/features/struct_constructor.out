GDTEST_OK
[test_int: 1, test_float: 2.5, test_dict: { 4: 5 }, test_array: [3, "Godot"]]
[test_int: 3, test_float: 2.5, test_dict: { 4: 5 }, test_array: [3, "Godot"]]
[test_int: 3, test_float: 1.5, test_dict: { 4: 5 }, test_array: [3, "Godot"]]
[test_int: 3, test_float: 1.5, test_dict: { "key": "val" }, test_array: [3, "Godot"]]
[test_int: 3, test_float: 1.5, test_dict: { "key": "val" }, test_array: [0.5]]
[name: "", class_name: &"", type: 0, hint: 0, hint_string: "", usage: 6]
[name: "", class_name: &"", type: 0, hint: 0, hint_string: "", usage: 6]
